WordPress/src/main/java/org/wordpress/android/ui/people/WPEditTextWithChipsOutlined.kt

```diff
@@ -0,0 +1,729 @@
+        NEUTRAL(R.attr.colorOnSurface),
+        VALIDATED(R.attr.colorPrimary),
+        VALIDATED_WITH_ERRORS(R.attr.wpColorError);
+
+        @ColorInt fun colorFromState(context: Context): Int {
+            val color = context.getColorResIdFromAttribute(this.colorAttr)
+
+            return ContextCompat.getColor(context, color)
+        }
+
+        companion object {
+            fun stateFromColor(context: Context, color: Int): ItemValidationState {
+                return values().first { value ->
+                    ContextCompat.getColor(context, context.getColorResIdFromAttribute(value.colorAttr)) == color
+                }
+            }
+        }
+    }
+
+    init {
+        attrs?.also {
+            val stylesAttributes = context.theme.obtainStyledAttributes(
+                    attrs,
+                    R.styleable.WPEditTextWithChipsOutlined,
+                    0,
+                    0
+            )
+
+            try {
+                hintResourceId = stylesAttributes.getResourceId(
+                        R.styleable.WPEditTextWithChipsOutlined_wpHint,
+                        0
+                )
+                chipifyEnabled = stylesAttributes.getBoolean(
+                        R.styleable.WPEditTextWithChipsOutlined_wpChipifyEnabled,
+                        false
+                )
+
+                maxChips = stylesAttributes.getInteger(
+                        R.styleable.WPEditTextWithChipsOutlined_wpMaxNumberChips, 0
+                )
+            } finally {
+                stylesAttributes.recycle()
+            }
+        }
+
+        inflate(getContext(), R.layout.wp_edit_text_with_chips_outlined, this)
+
+        flexBox = findViewById(R.id.flexbox)
+        editor = findViewById(R.id.flexbox_editor)
+        label = findViewById(R.id.flexbox_label)
+        hint = findViewById(R.id.flexbox_hint)
+
+        isRtl = RtlUtils.isRtl(context)
+
+        if (hasHint()) {
+            label.setText(hintResourceId)
+            hint.setText(hintResourceId)
+        }
+
+        isSaveEnabled = true
+
+        loadOutlineDrawable()
+        loadDimensions()
+        loadColors()
+
+        styleView(isEditorFocused(), hasItemsOrText(), false)
+
+        setListeners()
+    }
+
+    fun isEditorEmpty(): Boolean {
+        return !this.hasText()
+    }
+
+    fun getTextIfAvaliableOrNull(): String? {
+        throwExceptionIfChipifyNotEnabled()
+
+        if (!hasText() || !canAddMoreChips()) {
+            editor.setText("")
+            return null
+        }
+
+        return removeDelimiterFromItemIfPresent(editor.text.toString())
+    }
+
+    fun addOrUpdateChip(item: String, state: ItemValidationState) {
+        throwExceptionIfChipifyNotEnabled()
+        chipify(item, state)
+    }
+
+    fun removeChip(item: String) {
+        throwExceptionIfChipifyNotEnabled()
+        val chips = getChipsMap()
+
+        if (chips.containsKey(item)) {
```

Comment:
NIce! yep the idea could be to make the component cover also the case with duplicated chip items, but agree to change since now it's not covered 👍 . Changed here 507aca5cda1dfba582a342be69d8b8a433cbbff6